src/types/impl-trait.md

```diff
@@ -111,10 +133,193 @@ fn foo() -> impl Trait {
 doesn't allow the caller to determine the return type.
 Instead, the function chooses the return type, but only promises that it will implement `Trait`.
 
+
+## Type Alias Impl Trait
+
+In contrast to `impl Trait`s in function return types, type aliases can be used in more places than just return types and the same type alias can be used multiple times.
+
+```rust,ignore
```

Comment:
I'd generally prefer to avoid using `ignore` whenever possible, as it has caused problems in the past. Can you try to remove `ignore` from all of the examples? If there is some irrelevant parts needed, they can be prefixed with `#` to hide them. For example:

```
# trait Trait {}
# struct ImplementsTrait;
# impl Trait for ImplementsTrait {}

fn foo() -> impl Trait {
    # let value_of_type_that_implements_Trait = ImplementsTrait;
    value_of_type_that_implements_Trait
}
```